client/actions/auth.js

```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
   };
 }
 
+// calls checkSession to refresh idToken
+function refreshToken () {
+  return new Promise((resolve, reject) => {
+    // invoke check session to get a new token
+    webAuth.checkSession({},
+      function(err, result) {
+        if (err) { // there was an error
+          reject(err);
+        } else {  // we got a token
+          resolve(result);
+        }
+      }
+    );
+  });
+}
+
 const processTokens = (dispatch, apiToken, returnTo) => {
-  const decodedToken = jwtDecode(apiToken);
-  if (isExpired(decodedToken)) {
-    return;
-  }
 
-  axios.defaults.headers.common.Authorization = `Bearer ${apiToken}`;
-  axios.defaults.headers.common['dae-locale'] = window.config.LOCALE || 'en';
+ return new Promise((resolve, reject) => {
+    const decodedToken = jwtDecode(apiToken);
+    if (isExpired(decodedToken)) {
+      return;
+    }
 
-  sessionStorage.setItem('delegated-admin:apiToken', apiToken);
+    axios.defaults.headers.common.Authorization = `Bearer ${apiToken}`;
+    axios.defaults.headers.common['dae-locale'] = window.config.LOCALE || 'en';
+
+    sessionStorage.setItem('delegated-admin:apiToken', apiToken);
+
+    // creates an interceptor to refresh token if needed
+    axios.interceptors.request.use((config) => {
+      // get token from storage
+      const token = sessionStorage.getItem('delegated-admin:apiToken');
+      // if there is no token, or it is expired, try to get one
+      if(!token || isExpired(jwtDecode(token))){
```

Comment:
It might be nice to cache the expiresAt time so we aren't having to decode the token.  Also, when we "fix" this extension so it uses access tokens instead, we shouldn't be decoding the access token.